Lib/test/test_capi/test_opt.py

```
@@ -75,20 +75,19 @@ def loop():
                 self.assertEqual(opt.get_count(), 0)
                 with clear_executors(loop):
                     loop()
-                # Subtract because optimizer doesn't kick in sooner
-                self.assertEqual(opt.get_count(), 1000 - TIER2_THRESHOLD)
+                self.assertEqual(opt.get_count(), 1001)
```

I might be missing something. Why did we remove the piece about the `TIER2_THRESHOLD`?

I guess, more generally, how did you decide when to add/remove the threshold from a test? It seems we're wholesale replacing hardcoded values in tests but also adding/removing the threshold when basic arithmetic is used. Was this just trial and error, or is there some convention I'm not picking up on?

---

IIUC, we were iterating 1000 times,  `1000 - TIER2_THRESHOLD` of which happened after reaching the threshold.

By that doesn't work if `TIER2_THRESHOLD > 1000`, so now we iterate `1000 + TIER2_THRESHOLD` times, 1000 of which happen after reaching the threshold.

---

Earlier logic assumed a low threshold (< 1000). With the new threshold this logic needs to be adjusted to keep the test working. Now it's generic enough, that the threshold could be any (positive) number.

---

Ahhhh, that makes sense. Thanks!

---

Yeah, generally my approach was to change the number of loops to use the constant, then fix up any math to use the constant (for tests that assert some result). This one was a bit different, since it needed to run *more* than `TIER2_THRESHOLD` times. 1000 used to be a "big" number, but now it's not.